### PR TITLE
Add params support to associatedContact method for deals

### PR DESCRIPTION
**Pull request by sasanrose**

### CODE REVIEW

File: src/Resources/Deals.php [hunk 1]
```diff
@@ -138,13 +138,16 @@ function associateWithContact($dealId, $contactIds)
     
     /**
      * @param int $contactId
+     * @param array $params Optional parameters ['limit', 'offset']
      * @return mixed
      */
-    function associatedWithContact($contactId)
+    function associatedWithContact($contactId, array $params = [])
```
Comment (ryanwinchester) [inline src/Resources/Deals.php]: please don't include the array typehint 👍

Comment (sasanrose) [inline src/Resources/Deals.php]: But shouldn't it be consistent with other methods in the same class?

Comment (ryanwinchester) [inline src/Resources/Deals.php]: they should all go

Comment (sasanrose) [inline src/Resources/Deals.php]: Fair enough. Done

Comment (ryanwinchester) [inline src/Resources/Deals.php]: i haven't paid enouhh attention to all PRs, but I'll need to go through and remove array typehints at some point.

lots of people use collection objects, and other objects that have array access for things, but php doesn't really have a good typehint to cover those cases, unfortunately.

**Pull request merged**